File: fivem-server/resources/[rrp]/rrp-core/server/payday.lua
-- ============================================================
--  PAYDAY - se ruleaza la fiecare Config.Payday.intervalMinutes
--  Formula: brut = lvl*1000 + random(0..999)
--  Net = brut - tax_venit% - tax_cas% + bonus_factiune (suma1) + bonus_cass (suma2)
--  Bonusurile se aplica DOAR daca jucatorul are job intr-o factiune configurata
-- ============================================================

-- 'lvl' este interpretat ca level-ul curent al gradului in QBCore job, +1
local function getPlayerLevel(playerData)
    local grade = playerData.job and playerData.job.grade and playerData.job.grade.level or 0
    return math.max(1, grade + 1)
end

local function getConfig()
    local cfg = MySQL.single.await('SELECT tax_venit, tax_cas FROM rrp_payday_config WHERE id = 1') or { tax_venit = Config.Payday.defaultTaxVenit, tax_cas = Config.Payday.defaultTaxCAS }
    local pres = MySQL.single.await('SELECT suma1, suma2 FROM rrp_president WHERE id = 1') or { suma1 = Config.Payday.defaultSuma1, suma2 = Config.Payday.defaultSuma2 }
    return cfg.tax_venit, cfg.tax_cas, pres.suma1, pres.suma2
end

function RRP_RunPayday()
    local tax_venit, tax_cas, suma1, suma2 = getConfig()
    print(string.format('[PAYDAY] Tax venit=%d%% Tax CAS=%d%% Suma1=%d Suma2=%d', tax_venit, tax_cas, suma1, suma2))
    local count = 0
    for _, src in ipairs(GetPlayers()) do
        src = tonumber(src)
        local p = QBCore.Functions.GetPlayer(src)
        if p then
            local lvl = getPlayerLevel(p.PlayerData)
            local brut = lvl * 1000 + math.random(0, 999)
            local tv = math.floor(brut * tax_venit / 100)
            local tc = math.floor(brut * tax_cas / 100)
            local fac = RRP_PlayerHasFaction(p.PlayerData)
            local b1 = fac and suma1 or 0
            local b2 = fac and suma2 or 0
            local net = brut - tv - tc + b1 + b2
            p.Functions.AddMoney('bank', net, 'payday')

            MySQL.insert('INSERT INTO rrp_payday_history (citizenid, nume, salariu_brut, tax_venit, tax_cas, bonus_factiune, bonus_cass, salariu_net) VALUES (?,?,?,?,?,?,?,?)',
                { p.PlayerData.citizenid, p.PlayerData.name, brut, tv, tc, b1, b2, net })

            TriggerClientEvent('chat:addMessage', src, { color = {255, 214, 0}, args = {'[PAYDAY]', string.format('Brut $%d | -venit $%d | -CAS $%d | +fac $%d | +CASS $%d = NET $%d', brut, tv, tc, b1, b2, net)} })
            count = count + 1
        end
    end
    print('[PAYDAY] Procesati ' .. count .. ' jucatori online.')
end

-- Cron simplu in Lua
CreateThread(function()
    while true do
        Wait(Config.Payday.intervalMinutes * 60 * 1000)
        RRP_RunPayday()
    end
end)

-- Forteaza manual (doar admin lvl 10)
QBCore.Commands.Add('payday', 'Forteaza PayDay (admin 10)', {}, false, function(source)
    if RRP_GetAdminLevel(source) < 10 then
        return TriggerClientEvent('QBCore:Notify', source, '[Admin] Lvl 10 necesar.', 'error')
    end
    RRP_RunPayday()
    TriggerClientEvent('QBCore:Notify', source, 'PayDay rulat manual.', 'success')
end)

File: ragemp-server/packages/core/payday.js
// Sistem PayDay - se executa la fiecare ora fixa
// Formula: salariu_brut = lvl * 1000 + random(999)
// Aplica impozit pe venit, impozit pe CAS
// Daca este in factiune adauga bonus salariu (suma1) si bonus CASS (suma2)

const cron = require('node-cron');
const db = require('./database');
const players = require('./players');

async function getConfig() {
    const cfg = await db.one('SELECT * FROM payday_config WHERE id = 1');
    const pres = await db.one('SELECT suma1, suma2 FROM president WHERE id = 1');
    return {
        tax_venit: cfg ? cfg.tax_venit : 10,
        tax_cas: cfg ? cfg.tax_cas : 5,
        bonus_factiune: pres ? pres.suma1 : 2000,
        bonus_cass: pres ? pres.suma2 : 500
    };
}

async function calcularePentruJucator(p, cfg) {
    const lvl = p.lvl || 1;
    const salariu_brut = lvl * 1000 + Math.floor(Math.random() * 1000); // random(999) inclusiv 0..999
    const tax_venit_aplicat = Math.floor(salariu_brut * cfg.tax_venit / 100);
    const tax_cas_aplicat   = Math.floor(salariu_brut * cfg.tax_cas / 100);
    let bonus_factiune = 0;
    let bonus_cass = 0;
    if (p.factiune_id) {
        bonus_factiune = cfg.bonus_factiune;
        bonus_cass = cfg.bonus_cass;
    }
    const salariu_net = salariu_brut - tax_venit_aplicat - tax_cas_aplicat + bonus_factiune + bonus_cass;
    return { salariu_brut, tax_venit_aplicat, tax_cas_aplicat, bonus_factiune, bonus_cass, salariu_net };
}

async function ruleazaPayday() {
    const cfg = await getConfig();
    const allPlayers = await db.query('SELECT id, lvl, factiune_id FROM players');
    console.log(`[PAYDAY] Procesez ${allPlayers.length} jucatori. Tax venit=${cfg.tax_venit}% Tax CAS=${cfg.tax_cas}% Bonus fac=${cfg.bonus_factiune} CASS=${cfg.bonus_cass}`);

    for (const p of allPlayers) {
        const r = await calcularePentruJucator(p, cfg);
        await db.query('UPDATE players SET cont_bancar = cont_bancar + ?, ore_jucate = ore_jucate + 1 WHERE id = ?', [r.salariu_net, p.id]);
        await db.query(
            'INSERT INTO payday_history (player_id, salariu_brut, tax_venit_aplicat, tax_cas_aplicat, bonus_factiune, bonus_cass, salariu_net) VALUES (?,?,?,?,?,?,?)',
            [p.id, r.salariu_brut, r.tax_venit_aplicat, r.tax_cas_aplicat, r.bonus_factiune, r.bonus_cass, r.salariu_net]
        );
    }

    // Notifica jucatorii online (daca mp este disponibil)
    if (typeof mp !== 'undefined' && mp.players) {
        for (const online of players.getAllOnline()) {
            const playerRow = await db.one('SELECT lvl, factiune_id FROM players WHERE id = ?', [online.dbId]);
            if (!playerRow) continue;
            const r = await calcularePentruJucator(playerRow, cfg);
            // mesaj in chat - se itereaza jucatorii online via mp.players.toArray()
            for (const mpp of mp.players.toArray()) {
                if (players.getData(mpp)?.dbId === online.dbId) {
                    mpp.outputChatBox(`!{#FFD600}[PAYDAY] !{#FFFFFF}Salariu brut: $${r.salariu_brut} | Tax venit: -$${r.tax_venit_aplicat} | Tax CAS: -$${r.tax_cas_aplicat} | Bonus factiune: +$${r.bonus_factiune} | Bonus CASS: +$${r.bonus_cass}`);
                    mpp.outputChatBox(`!{#00E676}[PAYDAY] !{#FFFFFF}Net depus in cont: $${r.salariu_net}`);
                }
            }
        }
    }
}

function start() {
    // La fiecare ora fixa (minutul 0)
    cron.schedule('0 * * * *', async () => {
        try {
            await ruleazaPayday();
        } catch (e) {
            console.error('[PAYDAY] Eroare:', e);
        }
    });
    console.log('[PAYDAY] Scheduler pornit (ora fixa, 0 * * * *).');
}

module.exports = { start, ruleazaPayday, getConfig, calcularePentruJucator };
